From 6aae34c32e517d2c4c2a544fbdc1be74fbcd7cdb Mon Sep 17 00:00:00 2001
From: Fabian Hartung <fabian.hartung@chargebyte.com>
Date: Mon, 20 Jan 2025 08:27:01 +0100
Subject: [PATCH] Fix DCSupplySimulator initialization issue

This is only a temporary fix until the issue is fixed in the framework

Signed-off-by: Fabian Hartung <fabian.hartung@chargebyte.com>
---
 .../main/power_supply_DCImpl.hpp                 | 16 ++++++++--------
 1 file changed, 8 insertions(+), 8 deletions(-)

diff --git a/modules/simulation/DCSupplySimulator/main/power_supply_DCImpl.hpp b/modules/simulation/DCSupplySimulator/main/power_supply_DCImpl.hpp
index 0f847b2d..9d1df95a 100644
--- a/modules/simulation/DCSupplySimulator/main/power_supply_DCImpl.hpp
+++ b/modules/simulation/DCSupplySimulator/main/power_supply_DCImpl.hpp
@@ -61,15 +61,15 @@ private:
 
     // ev@3370e4dd-95f4-47a9-aaec-ea76f34a66c9:v1
     // insert your private definitions here
-    double settings_connector_export_voltage;
-    double settings_connector_import_voltage;
-    double settings_connector_max_export_current;
-    double settings_connector_max_import_current;
+    double settings_connector_export_voltage{0.0};
+    double settings_connector_import_voltage{0.0};
+    double settings_connector_max_export_current{0.0};
+    double settings_connector_max_import_current{0.0};
     types::power_supply_DC::Mode mode;
-    double connector_voltage;
-    double connector_current;
-    double energy_import_total;
-    double energy_export_total;
+    double connector_voltage{0.0};
+    double connector_current{0.0};
+    double energy_import_total{0.0};
+    double energy_export_total{0.0};
     std::mutex power_supply_values_mutex;
     Everest::Thread power_supply_thread_handle;
     types::powermeter::Powermeter power_meter_external();
-- 
2.34.1

